api/src/main/java/io/minio/MinioClient.java

```diff
@@ -4922,12 +5005,50 @@ public void setBucketNotification(
    * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
    * @throws XmlParserException thrown to indicate XML parsing error.
    */
+  @Deprecated
   public void removeAllBucketNotification(String bucketName)
       throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
-    NotificationConfiguration notificationConfiguration = new NotificationConfiguration();
-    setBucketNotification(bucketName, notificationConfiguration);
+    setBucketNotification(
+        SetBucketNotificationArgs.builder()
+            .bucket(bucketName)
+            .config(new NotificationConfiguration())
+            .build());
```

Comment:
Might be better to call `deleteBucketNotification` here

Reply:
Done